### PR TITLE
pbs_benchpress -> ptl_test_db.py JSONDecodeError

### PR DESCRIPTION
**Pull request by ericpershey**
#### Describe Bug or Feature
The JSON database was being overwritten, but the size of the file was
not being reduced, this was leaving garbage at the end of the file and
causing the JSON parser to fail.

#### Describe Your Change
Truncated the file to prevent the file from having leftover garbage.
The default type is set to "json", therefore corrected the "default"
in pbs_benchpress.  Also set __dbtype to None as the default is 
"json" and 'JSONDb' was not valid.

This is a fix for issue #1522

### CODE REVIEW

**Comment by toonen** [comment]
@agrawalravi90 I'm not sure what additional information you are looking for.  We did file an issue (#1522 ), which described the problem in detail, and then amended it with a test case to reproduce the problem.  The proposed fix in the PR is two lines to fix the problem (the calls to truncate) and three lines of other minor related issues.  What would you like us to explain further?

**Comment by agrawalravi90** [comment]
@toonen Sorry about the confusion, I was trying to give general information and not just specific to this PR. For this PR, can you just change the title of the PR and the commit to summarize the bug (maybe make it the same as the title of the Github Issue)?

**Comment by ericpershey** [comment]
I will change it shortly.  We have found another issue and a push is coming soon.  There is some kind of file race that still exists after those changes and takes longer to find.

File: test/fw/ptl/utils/plugins/ptl_test_db.py [hunk 1]
```diff
@@ -1615,16 +1615,19 @@ def __init__(self, dbtype, dbpath, dbaccess):
         self.res_data = PTLJsonData(command=self.__cmd)
 
     def __write_test_data(self, data):
-        jdata = None
         if _TESTRESULT_TN not in self.__dbobj.keys():
-            self.__dbobj[_TESTRESULT_TN] = open(self.dbpath, 'w+')
+            self.__dbobj[_TESTRESULT_TN] = self.dbpath
```
Comment (agrawalravi90) [inline test/fw/ptl/utils/plugins/ptl_test_db.py]: isn't self.__dbobj[_TESTRESULT_TN] redundant now? you already have the instance variable self.dbpath, and it seems like self.__dbobj[_TESTRESULT_TN] only gets set to one value: self.dbpath. Why not just do open("self.dbpath") instead in this class?

File: test/fw/ptl/utils/plugins/ptl_test_db.py [hunk 2]
```diff
@@ -1615,15 +1615,19 @@ def __init__(self, dbtype, dbpath, dbaccess):
...
-            jdata = json.load(self.__dbobj[_TESTRESULT_TN])
+            dbobj = self.__dbobj[_TESTRESULT_TN]
+            with open(dbobj, 'r') as fd:
+                jdata = json.load(fd)
         jsondata = self.res_data.get_json(data=data, prev_data=jdata)
-        self.__dbobj[_TESTRESULT_TN].seek(0)
-        json.dump(jsondata, self.__dbobj[_TESTRESULT_TN], indent=2)
+        with open(dbobj, 'w') as fd:
+            json.dump(jsondata, fd, indent=2)
+            fd.flush()
```
Comment (agrawalravi90) [inline test/fw/ptl/utils/plugins/ptl_test_db.py]: do you need the flush()? dump() also does a write(), so the next write() should just get buffered behind that, and all flushed when with block ends, right?

Comment (ericpershey) [reply test/fw/ptl/utils/plugins/ptl_test_db.py]: I think you are correct, I will remove the flush().

File: test/fw/ptl/utils/plugins/ptl_test_db.py (hunk 1 above)
Comment (ericpershey) [reply test/fw/ptl/utils/plugins/ptl_test_db.py]: Yes, it is redundant now.  I will remove self.__dbobj.

**Comment by ericpershey** [comment]
Updated, @agrawalravi90 can you take a look again.  Cleaned it up even more.  Also, how does the title, summary and issue look?

**Pull request merged**